### PR TITLE
Add tilde path tests

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Related to https://github.com/dotnet/coreclr/pull/16901.
Also clean up existing test data a bit.

Needs https://github.com/dotnet/coreclr/pull/16901 to pass. Marking as no merge for now.

cc: @danmosemsft, @pjanotti, @Anipik

### CODE REVIEW

File: src/System.Runtime.Extensions/tests/System/IO/PathTests.cs [hunk 1]
```diff
@@ -196,32 +196,73 @@ public void GetFullPath_InvalidArgs()
...
+                    { ".", currentDirectory },
+                    // ".." => up a directory
+                    { "..", Path.GetDirectoryName(currentDirectory) },
+                    // "dir/./././." => "dir"
+                    { Path.Combine(currentDirectory, ".", ".", ".", ".", "."), currentDirectory },
+                    // "dir///." => "dir"
+                    { currentDirectory + new string(Path.DirectorySeparatorChar, 3) + ".", currentDirectory },
+                    // "dir/../dir/./../dir" => "dir"
+                    { Path.Combine(currentDirectory, "..", fileName, ".", "..", fileName), currentDirectory },
+                    // "C:\somedir\.." => "C:\"
```
Comment (danmosemsft) [inline src/System.Runtime.Extensions/tests/System/IO/PathTests.cs]: Paths like `C: ...` are treated the same on Linux? I guess it's just string manipulation without regard to the OS?

Comment (JeremyKuhne) [reply src/System.Runtime.Extensions/tests/System/IO/PathTests.cs]: They aren't, it just is using the Windows example. This code is getting the root from the CWD, would be "/" on Unix.

File: src/System.Runtime.Extensions/tests/System/IO/PathTests.cs [hunk 2]
```diff
@@ -196,32 +196,73 @@ public void GetFullPath_InvalidArgs()
...
+            get
             {
-                longPath.Append(Path.DirectorySeparatorChar).Append('.');
+                // Paths with tildes '~' are processed for 8.3 expansion on Windows
+                string currentDirectory = Directory.GetCurrentDirectory();
+                string root = Path.GetPathRoot(currentDirectory);
+
+                TheoryData<string, string> data = new TheoryData<string, string>
+                {
+                    { "~", Path.Combine(currentDirectory, "~") },
```
Comment (danmosemsft) [inline src/System.Runtime.Extensions/tests/System/IO/PathTests.cs]: So GetFullPath("~") results in %CD%\~ ? How does that relate to `Paths with tildes '~' are processed for 8.3 expansion on Windows`. Are you just validating that `~` is not special (anymore?)

Comment (danmosemsft) [reply src/System.Runtime.Extensions/tests/System/IO/PathTests.cs]: comment maybe?

File: src/System.Runtime.Extensions/tests/System/IO/PathTests.cs (hunk 1 above)
Comment (danmosemsft) [reply src/System.Runtime.Extensions/tests/System/IO/PathTests.cs]: Ah yes.

File: src/System.Runtime.Extensions/tests/System/IO/PathTests.cs (hunk 2 above)
Comment (JeremyKuhne) [reply src/System.Runtime.Extensions/tests/System/IO/PathTests.cs]: Anything with a tilde gets passed to 8.3 expansion. We're just passing "normal" paths through that hit that code path to look for fall-on-our-face bugs. A file can be named `"~"` (although I wouldn't recommend it).

**Comment by JeremyKuhne** [comment]
@dotnet-bot test Windows x64 Debug Build 
@dotnet-bot test Windows x86 Release Build

**Comment by JeremyKuhne** [comment]
Still waiting on a new CoreCLR drop.

**Comment by JeremyKuhne** [comment]
@dotnet-bot test Windows x64 Debug Build
@dotnet-bot test Windows x86 Release Build

**Pull request merged**